mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -304,14 +305,21 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
-  @Since("2.0.0")
+  @Since("2.2.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+}
+  @Since("2.2.0")
```

Comment:
There is still no blank line above this line -- not sure how else to express this?
The indentation of the line above is off too.

Look, you should match the style you see around this code, and you can see how it isn't the same? just make it match.